lib/thrift/binary/framed/client.ex

```diff
@@ -252,8 +211,8 @@ defmodule Thrift.Binary.Framed.Client do
       {:error, :timeout} = timeout ->
         {:reply, timeout, s}
 
-      {:error, :closed} = err ->
-        {:disconnect, {:retry, err}, %{s |last_message: {call_args, caller}}}
```

Comment:
`{:error, :closed}` used to be special case so that it would always trigger a reconnect attempt - however if `retry: false` (default) the process is stopped immediately after setting up a fresh connection https://github.com/pinterest/elixir-thrift/pull/303/files#diff-217ed11585237b96d356fba460274021L144 and a reply would never be sent. I think this is a bug in the test/previous behavior.